comp/rdnsquerier/impl/config.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package rdnsquerierimpl
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+)
+
+type rdnsQuerierConfig struct {
+	enabled  bool
+	workers  int
+	chanSize int
+
+	rateLimiterEnabled bool
+	rateLimiterLimit   int
+	rateLimiterBurst   int
+
+	cacheEnabled         bool
+	cacheEntryTTL        int
+	cacheCleanInterval   int
+	cachePersistInterval int
+
+	// debug - TODO remove
+	fakeResolver                 bool
+	generateFakeQueriesPerSecond int
+	lookupDelayMs                int
+}
+
+func newConfig(agentConfig config.Component, logger log.Component) *rdnsQuerierConfig {
+	netflowRDNSEnrichmentEnabled := agentConfig.GetBool("network_devices.netflow.reverse_dns_enrichment_enabled")
+
+	c := &rdnsQuerierConfig{
+		enabled:  netflowRDNSEnrichmentEnabled,
+		workers:  agentConfig.GetInt("reverse_dns_enrichment.workers"),
+		chanSize: agentConfig.GetInt("reverse_dns_enrichment.chan_size"),
+
+		rateLimiterEnabled: agentConfig.GetBool("reverse_dns_enrichment.rate_limiter.enabled"),
+		rateLimiterLimit:   agentConfig.GetInt("reverse_dns_enrichment.rate_limiter.limit"),
+		rateLimiterBurst:   agentConfig.GetInt("reverse_dns_enrichment.rate_limiter.burst"),
+
+		cacheEnabled:         agentConfig.GetBool("reverse_dns_enrichment.cache.enabled"),
+		cacheEntryTTL:        agentConfig.GetInt("reverse_dns_enrichment.cache.entry_ttl"),
```

Comment:
Ah I didn't realize we were using plain int here :/ Better to aim for consistency I'd say but no strong opinion now

Reply:
I decided to use time.Duration since the reverse_dns_enrichment config is separate from the network_device.netflow config - commit f7b077d5ffded873c49e37202eb067ec6c8ba4c3.